### DISCUSSION TITLE
LaTeX/MathJax rendering bugs

### DISCUSSION THREAD

**Discussion by ilemhadri** [community/community#36915] [Enterprise]
Reading through this https://github.com/orgs/community/discussions/18897, it appears there were other bugs in the past and Github has fixed them.

This is an example of very simple laTeX expression that is still not rendering correctly in either Github or Github Enterprise
`$\bar{X}_{i,m} = X \Phi_{i,m}^T$`
whereas both `$\bar{X}_{i,m} = $` and `$X \Phi_{i,m}^T$` render properly.

This kind of issues are unfortunately blocking us from being confident in rolling out LaTeX usage more broadly in Markdown in Github.

**Comment by ossanna16** [comment]
Hi there @ilemhadri and welcome to our community 😄. Thanks so much for asking a great question!

**Comment by dpvc** [comment]
The GitHub implementation of MathJax has a number of problems.  Because LaTeX notation doesn't mesh well with Markdown notation, they have tried to handle that by quoting the LaTeX characters in ways that would prevent them from being processed by Markdown.  Unfortunately, they haven't gotten in right, and there are a number of situations that they have not handled properly.  One of them is underscores, which Markdown uses to indicate italics.  So when you type

    $\bar{X}_{i,m} = X \Phi_{i,m}^T$

you get

$\bar{X}_{i,m} = X \Phi_{i,m}^T$

Note that the text between the underscores is in italics and the underscores are missing, which means Markdown has replaced the underscores with HTML tags that indicate italics.  MathJax doesn't process math that includes HTML tags, so that prevents MathJax from typesetting the result.

Your example seems to not have the underscores at all, so it may be that when an item is posted, the processing is different (I will find out shortly when this is posted), but in the preview the math is not typeset due to the HTML tags for italic.

You should be able to get it to work using 

    $\bar{X}\_{i,m} = X \Phi\_{i,m}^T$

which produces

$\bar{X}\_{i,m} = X \Phi\_{i,m}^T$

though if they ever fix this problem, your equations will end up with actual underscores in the output, like

$\bar{X}\\\_{i,m} = X \Phi\\\_{i,m}^T$

**Reply by nschloe** [reply]
I wish they had taken inspiration from GitHub's math support which uses backticks to protect the math content as "code",

````markdown
$`a^2 + b^2 = c^2`$
````

They do allow backticked block math now, which fixed all of the problems there,

````markdown
```math
\exp(i\tau) - 1 = 0
```
````
but inline math remains a mess.

**Comment by nschloe** [comment]
Added this to my math bug overview. https://github.com/nschloe/github-math-bugs

**Comment by nschloe** [comment]
This is fixed by using the new inline math backtick delimiters:

https://github.blog/changelog/2023-05-08-new-delimiter-syntax-for-inline-mathematical-expressions/

> $`\bar{X}_{i,m} = X \Phi_{i,m}^T`$

**Comment by RGafiyatullin** [comment]
For some reason, I couldn't find out, certain math expressions are not rendered (left as plain text).

[Demo](https://gist.github.com/RGafiyatullin/37a1676b6b3ceeeee2ffc1f1ad2ad00e)

**Comment by thehappycheese** [comment]
I found a bug related to pre-processing underscores probably...




### Not working example

```tex
$$
\begin{align*}
\text{Test}_b &= x_1\\
q_{y+1} &= q
\end{align*}
$$
```

Does not render:

$$
\begin{align*}
\text{Test}_b &= x_1\\
q_{y+1} &= q
\end{align*}
$$

I see this in the preview:

![image](https://github.com/community/community/assets/2748120/d44b18e0-4e20-403d-8b21-383b72b1ea00)

### Make it work using hack

But this does working using a hack that breaks other latex viewers
(I inserted a `\` before the first `_` in the second equaiton

```tex
$$
\begin{align*}
\text{Test}_b &= x_1\\
q\_{y+1} &= q
\end{align*}
$$
```

$$
\begin{align*}
\text{Test}_b &= x_1\\
q\_{y+1} &= q
\end{align*}
$$

I see this in the preview:

![image](https://github.com/community/community/assets/2748120/5a628932-8037-4671-8401-ff2e24575acf)

**Comment by henrygerardmoore** [comment]
**Edit: See bottom of this post, looks like this is just a browser issue for me**

It looks like vertical matrices and variable-height braces in don't work, see below:

```math
\begin{bmatrix} x \\ y \\ z \end{bmatrix}
```

Inline also doesn't work: $`\begin{bmatrix} x \\ y \\ z \end{bmatrix}`$


Here's an example without using `bmatrix`:
```
\left( (\mathrm{really} ^ {\mathrm{tall} ^ \mathrm{example}} _ {\mathrm{in}_{\mathrm{both}_{directions}}})\right)
```

```math
\left( (\mathrm{really} ^ {\mathrm{tall} ^ \mathrm{example}} _ {\mathrm{in}_{\mathrm{both}_{directions}}})\right)
```

Here's what it looks like when previewed in VSCode's markdown preview:
![image](https://github.com/user-attachments/assets/c92d30db-1c09-4579-80a5-38ce66125493)


And in the github preview:
![image](https://github.com/user-attachments/assets/ecf887be-c88a-40d5-9252-7b775759904d)

Notice that the `\left(` and `\right)` parentheses are identical to the regular ones on github

Edit:
This is on `Google Chrome 130.0.6723.91`
From a coworker, looks like this renders as expected on `129.0.6668.100`. I tried on firefox and it also renders correctly, so this is just a latest version google chrome issue as far as I can tell.
